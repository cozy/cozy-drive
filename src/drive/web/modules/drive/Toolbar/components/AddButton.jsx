import React, { useContext } from 'react'

import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import Button from 'cozy-ui/transpiled/react/Button'
import PlusIcon from 'cozy-ui/transpiled/react/Icons/Plus'

import { AddMenuContext } from 'drive/web/modules/drive/AddMenu/AddMenuProvider'

export const AddButton = () => {
  const { t } = useI18n()
  const {
    anchorRef,
    handleToggle,
    isDisabled,
    handleOfflineClick,
    isOffline,
    a11y
  } = useContext(AddMenuContext)

  return (
    <div ref={anchorRef} onClick={isOffline ? handleOfflineClick : undefined}>
      <Button
        onClick={handleToggle}
        disabled={isDisabled || isOffline}
        icon={PlusIcon}
        label={t('toolbar.menu_add')}
        {...a11y}
      />
    </div>
  )
}

export default React.memo(AddButton)
