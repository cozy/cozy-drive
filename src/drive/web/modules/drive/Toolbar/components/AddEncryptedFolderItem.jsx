import React from 'react'
import { connect } from 'react-redux'

import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
import { ActionMenuItem } from 'cozy-ui/transpiled/react/deprecated/ActionMenu'
import EncryptedFolderIcon from 'drive/web/modules/views/Folder/EncryptedFolderIcon'
import { useVaultUnlockContext } from 'cozy-keys-lib'
import {
  showNewFolderInput,
  encryptedFolder
} from 'drive/web/modules/filelist/duck'

const AddEncryptedFolderItem = ({ addEncryptedFolder }) => {
  const { t } = useI18n()
  const { showUnlockForm } = useVaultUnlockContext()

  return (
    <ActionMenuItem
      data-testid="add-encrypted-folder-link"
      onClick={() => showUnlockForm({ onUnlock: addEncryptedFolder })}
      left={<EncryptedFolderIcon width={16} height={16} />}
    >
      {t('toolbar.menu_new_encrypted_folder')}
    </ActionMenuItem>
  )
}

const mapDispatchToProps = dispatch => ({
  // An encrypted folder triggers a new encryption key generation
  addEncryptedFolder: () => {
    dispatch(encryptedFolder())
    dispatch(showNewFolderInput())
  }
})

export default connect(null, mapDispatchToProps)(AddEncryptedFolderItem)
