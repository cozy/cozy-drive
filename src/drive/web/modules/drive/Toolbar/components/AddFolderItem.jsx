import React from 'react'
import { connect } from 'react-redux'
import { translate } from 'cozy-ui/react/I18n'
import styles from 'drive/styles/toolbar'
import { showNewFolderInput } from 'drive/web/modules/filelist/duck'

const AddFolderItem = translate()(({ t, addFolder }) => {
  return (
    <a
      data-test-id="add-folder-link"
      className={styles['fil-action-newfolder']}
      onClick={addFolder}
    >
      {t('toolbar.menu_new_folder')}
    </a>
  )
})
const mapDispatchToProps = dispatch => ({
  addFolder: () => dispatch(showNewFolderInput())
})

export default connect(
  null,
  mapDispatchToProps
)(AddFolderItem)
