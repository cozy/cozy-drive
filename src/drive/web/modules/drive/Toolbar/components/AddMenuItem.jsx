import React, { useContext } from 'react'

import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
import Icon from 'cozy-ui/transpiled/react/Icon'
import PlusIcon from 'cozy-ui/transpiled/react/Icons/Plus'
import { AddMenuContext } from 'drive/web/modules/drive/AddMenu/AddMenuProvider'

const AddMenuItem = () => {
  const { t } = useI18n()

  const {
    anchorRef,
    handleToggle,
    isDisabled,
    handleOfflineClick,
    isOffline,
    a11y
  } = useContext(AddMenuContext)

  return (
    <div ref={anchorRef} onClick={isOffline ? handleOfflineClick : undefined}>
      <ActionMenuItem
        left={<Icon icon={PlusIcon} />}
        disabled={isDisabled || isOffline}
        icon={PlusIcon}
        onClick={handleToggle}
        {...a11y}
      >
        {t('toolbar.menu_add_item')}
      </ActionMenuItem>
    </div>
  )
}

export default AddMenuItem
