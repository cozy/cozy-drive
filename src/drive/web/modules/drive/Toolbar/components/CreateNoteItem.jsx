import React from 'react'
import get from 'lodash/get'

import { translate } from 'cozy-ui/transpiled/react/I18n'
import {
  withClient,
  models,
  useAppLinkWithStoreFallback,
  useCapabilities
} from 'cozy-client'
import { generateUniversalLink } from 'cozy-ui/transpiled/react/AppLinker/native'
import { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
import Icon from 'cozy-ui/transpiled/react/Icon'

import toolbarContainer from 'drive/web/modules/drive/Toolbar/toolbar'

const CreateNoteItem = ({ client, t, displayedFolder }) => {
  const capabilities = useCapabilities(client)
  const isFlatDomain = get(
    capabilities,
    'capabilities.data.attributes.flat_subdomains'
  )

  let notesAppUrl = undefined
  let notesAppIsInstalled = true

  const { fetchStatus, url, isInstalled } = useAppLinkWithStoreFallback(
    'notes',
    client
  )
  if (fetchStatus === 'loaded') {
    notesAppUrl = url
    notesAppIsInstalled = isInstalled
  }

  let returnUrl = ''
  if (displayedFolder) {
    returnUrl = generateUniversalLink({
      slug: 'drive',
      cozyUrl: client.getStackClient().uri,
      subDomainType: isFlatDomain ? 'flat' : 'nested',
      nativePath: `/files/${displayedFolder.id}`
    })
  }

  return (
    <ActionMenuItem
      data-test-id="create-a-note"
      left={<Icon icon="note" />}
      onClick={async () => {
        if (notesAppUrl === undefined) return
        if (notesAppIsInstalled) {
          const { data: file } = await client.create('io.cozy.notes', {
            dir_id: displayedFolder.id
          })

          window.location.href = await models.note.generatePrivateUrl(
            notesAppUrl,
            file,
            { returnUrl }
          )
        } else {
          window.location.href = notesAppUrl
        }
      }}
    >
      {t('toolbar.menu_create_note')}
    </ActionMenuItem>
  )
}

export default translate()(withClient(toolbarContainer(CreateNoteItem)))
