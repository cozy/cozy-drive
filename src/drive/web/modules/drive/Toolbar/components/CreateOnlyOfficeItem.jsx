import React, { useCallback, useMemo } from 'react'
import get from 'lodash/get'

import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
import Icon from 'cozy-ui/transpiled/react/Icon'

import { ROOT_DIR_ID } from 'drive/constants/config'
import { useRouter } from 'drive/lib/RouterContext'
import {
  makeOnlyOfficeIconByClass,
  canWriteOfficeDocument
} from 'drive/web/modules/views/OnlyOffice/helpers'

const CreateOnlyOfficeItem = ({ fileClass }) => {
  const { t } = useI18n()
  const { router } = useRouter()

  const folderId = useMemo(
    () => get(router, 'params.folderId', ROOT_DIR_ID),
    [router]
  )

  const handleClick = useCallback(() => {
    if (canWriteOfficeDocument()) {
      router.push(`/onlyoffice/create/${folderId}/${fileClass}`)
    } else {
      router.push(`/folder/${folderId}/paywall`)
    }
  }, [router, fileClass, folderId])

  const ClassIcon = useMemo(
    () => makeOnlyOfficeIconByClass(fileClass),
    [fileClass]
  )

  return (
    <ActionMenuItem onClick={handleClick} left={<Icon icon={ClassIcon} />}>
      {t(`toolbar.menu_onlyOffice.${fileClass}`)}
    </ActionMenuItem>
  )
}

export default React.memo(CreateOnlyOfficeItem)
