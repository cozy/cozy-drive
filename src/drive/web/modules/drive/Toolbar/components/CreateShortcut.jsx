import React, { useState, useCallback } from 'react'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import { ActionMenuItem } from 'cozy-ui/transpiled/react/ActionMenu'
import Icon from 'cozy-ui/transpiled/react/Icon'
import ShortcutCreationModal from './ShortcutCreationModalConnected'

const CreateShortcutWrapper = () => {
  const [isModalDisplayed, setIsModalDisplayed] = useState(false)
  const { t } = useI18n()
  if (!isModalDisplayed) {
    return (
      <ActionMenuItem
        data-test-id="create-a-shortcut"
        left={<Icon icon="link" />}
        onClick={useCallback(() => {
          setIsModalDisplayed(true)
        })}
      >
        {t('toolbar.menu_create_shortcut')}
      </ActionMenuItem>
    )
  } else {
    return (
      <ShortcutCreationModal
        onClose={useCallback(() => setIsModalDisplayed(false))}
      />
    )
  }
}

export default CreateShortcutWrapper
