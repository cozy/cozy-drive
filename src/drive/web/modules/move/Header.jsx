import React from 'react'
import PropTypes from 'prop-types'
import {
  Well,
  Media,
  Img,
  Bd,
  Circle,
  Counter,
  Icon,
  Button,
  Caption,
  Bold
} from 'cozy-ui/react'
import palette from 'cozy-ui/stylus/settings/palette.json'
import DriveIcon from 'drive/assets/icons/icon-drive.svg'

const HeaderIlustration = ({ count }) => {
  return count > 1 ? (
    <Circle>
      <Counter count={count} />
    </Circle>
  ) : (
    <Icon icon={DriveIcon} size={32} />
  )
}

const Header = ({ entries, onClose }, { t }) => (
  <Well>
    <Media>
      <Img className="u-mr-1">
        <HeaderIlustration count={entries.length} />
      </Img>
      <Bd>
        <Bold ellipsis>
          {entries.length > 1
            ? t('Move.title', { smart_count: entries.length })
            : entries[0].name}
        </Bold>
        <Caption ellipsis>{t('Move.to')}</Caption>
      </Bd>
      <Button theme="close" onClick={onClose} extension="narrow" type="button">
        <Icon icon="cross" width="14" height="14" color={palette['coolGrey']} />
      </Button>
    </Media>
  </Well>
)

Header.propTypes = {
  entries: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string
    })
  ).isRequired,
  onClose: PropTypes.func.isRequired
}

Header.contextTypes = {
  t: PropTypes.func.isRequired
}

export default Header
