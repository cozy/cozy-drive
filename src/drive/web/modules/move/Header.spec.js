import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'
import { TestI18n } from 'test/components/AppLike'
import en from 'drive/locales/en.json'
import Polyglot from 'node-polyglot'
const onCloseSpy = jest.fn()
describe('Header', () => {
  const setupComponent = ({ entries = [], onClose, title, subTitle }) => {
    const props = {
      entries,
      onClose,
      title,
      subTitle
    }
    const polyglot = new Polyglot()
    polyglot.extend(en)
    const t = polyglot.t.bind(polyglot)
    return {
      ...render(
        <TestI18n>
          <Header {...props} />
        </TestI18n>
      ),
      t
    }
  }
  it('should fallback to Move title if no title is given', () => {
    const { getByText, t } = setupComponent({ onClose: onCloseSpy })
    expect(getByText(t('Move.title', { smart_count: 0 })))
  })
  it('should display title if title is given and no file', () => {
    const { getByText } = setupComponent({
      onClose: onCloseSpy,
      title: 'My Title'
    })
    expect(getByText('My Title'))
  })
  it('should display the right title if only one file', () => {
    const { getByText } = setupComponent({
      onClose: onCloseSpy,
      title: 'My Title',
      entries: [
        {
          name: 'FileName.txt'
        }
      ]
    })
    expect(getByText('FileName.txt'))
    //expect(component).toMatchSnapshot()
  })
  it('should display the right title if more than one files', () => {
    const { getByText } = setupComponent({
      onClose: onCloseSpy,
      title: 'My Title',
      entries: [
        {
          name: 'FileName.txt'
        },
        {
          name: 'FileName2.txt'
        }
      ]
    })
    expect(getByText('My Title'))
  })
})
