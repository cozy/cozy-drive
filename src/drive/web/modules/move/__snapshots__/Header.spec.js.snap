// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should display the right title if more than one files 1`] = `
<Well>
  <Media>
    <Img
      className="u-mr-1"
    >
      <HeaderIlustration
        count={2}
      />
    </Img>
    <Bd>
      <ForwardRef(Bold)
        ellipsis={true}
        tag="div"
      >
        My Title
      </ForwardRef(Bold)>
      <ForwardRef(Caption)
        ellipsis={true}
        tag="div"
      >
        Move.to
      </ForwardRef(Caption)>
    </Bd>
    <Button
      align="center"
      extension="narrow"
      onClick={[MockFunction]}
      size="normal"
      tag="button"
      theme="close"
      type="button"
    >
      <Icon
        color="var(--coolGrey)"
        height="14"
        icon="cross"
        spin={false}
        width="14"
      />
    </Button>
  </Media>
</Well>
`;

exports[`Header should display the right title if only one file 1`] = `
<Well>
  <Media>
    <Img
      className="u-mr-1"
    >
      <HeaderIlustration
        count={1}
      />
    </Img>
    <Bd>
      <ForwardRef(Bold)
        ellipsis={true}
        tag="div"
      >
        FileName.txt
      </ForwardRef(Bold)>
      <ForwardRef(Caption)
        ellipsis={true}
        tag="div"
      >
        Move.to
      </ForwardRef(Caption)>
    </Bd>
    <Button
      align="center"
      extension="narrow"
      onClick={[MockFunction]}
      size="normal"
      tag="button"
      theme="close"
      type="button"
    >
      <Icon
        color="var(--coolGrey)"
        height="14"
        icon="cross"
        spin={false}
        width="14"
      />
    </Button>
  </Media>
</Well>
`;

exports[`Header should display title if title is given and no file 1`] = `
<Well>
  <Media>
    <Img
      className="u-mr-1"
    >
      <HeaderIlustration
        count={0}
      />
    </Img>
    <Bd>
      <ForwardRef(Bold)
        ellipsis={true}
        tag="div"
      >
        My Title
      </ForwardRef(Bold)>
      <ForwardRef(Caption)
        ellipsis={true}
        tag="div"
      >
        Move.to
      </ForwardRef(Caption)>
    </Bd>
    <Button
      align="center"
      extension="narrow"
      onClick={[MockFunction]}
      size="normal"
      tag="button"
      theme="close"
      type="button"
    >
      <Icon
        color="var(--coolGrey)"
        height="14"
        icon="cross"
        spin={false}
        width="14"
      />
    </Button>
  </Media>
</Well>
`;

exports[`Header should fallback to Move title if no title is given 1`] = `
<Well>
  <Media>
    <Img
      className="u-mr-1"
    >
      <HeaderIlustration
        count={0}
      />
    </Img>
    <Bd>
      <ForwardRef(Bold)
        ellipsis={true}
        tag="div"
      >
        Move.title
      </ForwardRef(Bold)>
      <ForwardRef(Caption)
        ellipsis={true}
        tag="div"
      >
        Move.to
      </ForwardRef(Caption)>
    </Bd>
    <Button
      align="center"
      extension="narrow"
      onClick={[MockFunction]}
      size="normal"
      tag="button"
      theme="close"
      type="button"
    >
      <Icon
        color="var(--coolGrey)"
        height="14"
        icon="cross"
        spin={false}
        width="14"
      />
    </Button>
  </Media>
</Well>
`;
